src/project/internal/notationproject.cpp

```diff
@@ -641,10 +655,39 @@ Ret NotationProject::doSave(const muse::io::path_t& path, engraving::MscIoMode i
                 LOGW() << ret.toString();
             }
         } else {
-            Ret ret = fileSystem()->move(savePath, targetContainerPath, true);
-            if (!ret) {
-                return ret;
+            Ret ret = muse::make_ok();
+
+            if (!targetContainerPath.contains(" - ALL_ZEROS_CORRUPTED_ORIG_ONLY.mscz")) {
+                ret = fileSystem()->copy(savePath, targetContainerPath, true);
+                if (!ret) {
+                    return ret;
+                }
+            } else {
+                // Corrupt the original file for testing purposes. The temp file will be healthy.
+                size_t tempFileSize = QFileInfo(savePath).size();
+                ByteArray corruptedData = ByteArray(tempFileSize);
+                ret = AllZerosFileCorruptor::writeFile(targetContainerPath, corruptedData);
+                if (!ret) {
+                    return ret;
+                }
+            }
+
+            if (!isAutosave) {
+                ret = checkSavedFileForCorruption(ioMode, targetContainerPath, targetMainFileName.toQString());
+                if (!ret) {
+                    if (ret.code() == (int)Err::CorruptionUponSavingError) {
+                        // Validate the temporary "saving" file too.
+                        Ret ret2 = checkSavedFileForCorruption(ioMode, savePath, targetMainFileName.toQString());
```

Comment:
Looks like ret2 is redundant here (as well as the check below):

```suggestion
                        ret = checkSavedFileForCorruption(ioMode, savePath, targetMainFileName.toQString());
```